include/sqls_rtti_comp.hpp

```diff
@@ -130,23 +130,8 @@ struct LesserRTTI
     return false;
   }
 
-  __host__ __device__
-  void gather(void** d_ppcols_out, size_t* d_indices, size_t nrows_new) const

```

Comment:
I do not see LesserRTTI::gather being used anywhere in the repository. I think it should be same to remove them.

Reply:
*that* gather() not only was not used anywhere, but it also invited the possibility of dynamic parallelism, which is *very* expensive. In the old code Gatherer::operator() was __host__ __device__ qualified and it was calling that version of gather that had thrust::device exe space. That opened the possibility of calling a kernel from inside a kernel; i.e., dynamic parallelism, which can be 2 orders of magnitude slower.

The main reason for trimming the __host__ __device__ qualifiers in that code is to avoid this.